esmvaltool/namelist.py

```diff
@@ -365,12 +365,32 @@ def _update_multi_model_stats(variables, settings):
             settings['multi_model_stats'] = {}
         settings['multi_model_stats']['filename'] = {}
         variable = variables[0]
+        #####################################
+        # Notes on multimodel stats file name
+        # delete them after decistion
+        #####################################
+        # BIG VARIABLE DICTY-DICT example
+        # {'preprocessor': 'pp850', 'reference_model': 'ERA-Interim',
+        # 'alternative_model': 'NCEP', 'mip': 'Amon', 'field': 'T3M',
+        # 'short_name': 'ta', 'diagnostic': 'ta850glob', 'model':
+        # 'bcc-csm1-1', 'project': 'CMIP5', 'exp': 'historical',
+        # 'ensemble': 'r1i1p1', 'start_year': 2000, 'end_year': 2002,
+        # 'cmor_table': 'CMIP5', 'standard_name': 'air_temperature',
+        # 'long_name': 'Air Temperature', 'units': 'K', 'filename':
+        # 'blahblah.nc'}
+        # typical data file name (CMIP5)
+        # ta_Amon_MPI-ESM-LR_historical_r1i1p1_200001-200512.nc
+        # build multimodel stats file name as like
+        # [short_name]_[preprocessor]_BACKEND-MultiModel_[stats].nc
```

Comment:
I would remove `_` before `[stats]`. The reason is that `MultiModelMean` and `MultiModelMean` will be added as models name to the models list and `_` is not allowed in names since it is already used as separator in the file name.

I would also suggest to name the output file mimicking the CMIP5 `output_file` as defined in `config-developer.yml`. Hence my original suggestion:

```
BACKEND_MultiModel[stats]_[field]_[var]_[start_year]-[end_year].nc
```
i.e., without `[mip]`, `[exp]` and [ensemble]` which are not always available for all models. 

`[preprocessor]` is a redundant information here, since the directory where the file is saved already contains this information in the path.

Reply:
@mattiarighi good points! I really have no preference how they're named, by me M1A1 Abrams would be a valid name too :)) Only thing about the years is that variable = variables[0] so the first model in the list, it could be that other models have longer/shorter time spans and this way the name is not 100% accurate on time, depending what time gating option we use in multi_model_stats (overlap or full).

Reply:
`[start_year]` and `[end_year]` must be consistent with the chosen option `overlap` or `full`